src/routes/ProtectedRoute.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect, useState } from 'react';
+import { withAuthenticationRequired, useAuth0 } from '@auth0/auth0-react';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+const ProtectedRoute = (component) => (props) => {
+  const { isAuthenticated, getAccessTokenSilently } = useAuth0();
+  let token;
+  const [authStatus, setAuthStatus] = useState(null);
```

Comment:
Why did you have to use a separate way to keep auth status from Auth0's is authenticated?

Reply:
Because `isAuthenticated` is boolean and I had three states: 
1. Authentication is being fetched
2. Is not authenticated
3. is authenticated